lib/documents.js

```diff
@@ -164,6 +168,56 @@ export async function register({
   }
 }
 
+/**
+ * Encrypts the document either once (with the given recipients) or recursively
+ * in a chain (using the recipientChain).
+ *
+ * @param {object} options - Options hashmap.
+ * @param {object} options.document - The document to register.
+ * @param {Array<string>} recipients - See `register()` docstring.
+ * @param {Array<Array<string>>} recipientChain - See `register()` docstring.
+ *
+ * @returns {Promise<object>} Resolves with the JWE, the result of encryption.
+ */
+export async function encrypt({document, recipients, recipientChain}) {
+  if(!(recipients || recipientChain)) {
+    throw new BedrockError(
+      'Encryption recipients or recipientChain is required.',
+      'ValidationError', {
+        public: true,
+        httpStatusCode: 400
+      });
+  }
+
+  const obj = document;
+  const keyResolver = createKeyResolver();
+  const cipher = new Cipher();
+
+  if(recipientChain) {
+    if(!Array.isArray(recipientChain)) {
+      throw new BedrockError(
+        '"recipientChain" must be an array.',
+        'ValidationError', {
+          public: true,
+          httpStatusCode: 400
+        });
+    }
+
+    const [currentRecipients, ...remainingRecipientChain] = recipientChain;
+    const jwe = await cipher.encryptObject({
+      obj, recipients: currentRecipients, keyResolver
+    });
```

Comment:
Can we arrange this as a loop instead of using recursion? If `recipients` are given, we can normalize it to a `recipientsChain` with just 1 entry and then do a loop over it. Loops are easier to reason about, easier to debug, don't require us to consider stack depth, so on.

Reply:
Sure, switched to iteration.